From d7edf79ddb10e650ea7c285b67497a9d18e2e145 Mon Sep 17 00:00:00 2001
From: Stanislav Poboril <stanislav.poboril@nxp.com>
Date: Wed, 7 Nov 2018 13:13:57 +0100
Subject: [PATCH] arm: imx: disable mcc and mu driver support

Disabled MCC and MU driver support in mach-imx so that MU could be
replaced by an alternative version.

Upstream-Status: Inappropriate [disable feature]

Signed-off-by: Stanislav Poboril <stanislav.poboril@nxp.com>
---
 arch/arm/configs/udoo_neo_defconfig | 4 ++--
 arch/arm/mach-imx/Kconfig           | 1 -
 arch/arm/mach-imx/Makefile          | 2 +-
 arch/arm/mach-imx/pm-imx6.c         | 4 +---
 4 files changed, 4 insertions(+), 7 deletions(-)

diff --git a/arch/arm/configs/udoo_neo_defconfig b/arch/arm/configs/udoo_neo_defconfig
index d9b87fd..908c45b 100644
--- a/arch/arm/configs/udoo_neo_defconfig
+++ b/arch/arm/configs/udoo_neo_defconfig
@@ -314,7 +314,7 @@ CONFIG_HAVE_IMX_ANATOP=y
 CONFIG_HAVE_IMX_GPC=y
 CONFIG_HAVE_IMX_MMDC=y
 CONFIG_HAVE_IMX_SRC=y
-CONFIG_HAVE_IMX_MCC=y
+CONFIG_HAVE_IMX_MCC=n
 CONFIG_HAVE_IMX_AMP=y
 
 #
@@ -1597,7 +1597,7 @@ CONFIG_HW_RANDOM=m
 # CONFIG_TCG_TPM is not set
 CONFIG_IMX_SEMA4=y
 # CONFIG_IMX_MCC_TEST is not set
-CONFIG_IMX_UDOONEO_MCC=y
+CONFIG_IMX_UDOONEO_MCC=n
 CONFIG_I2C=y
 CONFIG_I2C_BOARDINFO=y
 # CONFIG_I2C_COMPAT is not set
diff --git a/arch/arm/mach-imx/Kconfig b/arch/arm/mach-imx/Kconfig
index 8636d68..3253259 100644
--- a/arch/arm/mach-imx/Kconfig
+++ b/arch/arm/mach-imx/Kconfig
@@ -839,7 +839,6 @@ config SOC_IMX6SX
 	bool "i.MX6 SoloX support"
 	select PINCTRL_IMX6SX
 	select HAVE_IMX_AMP
-	select HAVE_IMX_MCC
 	select SOC_IMX6
 
 	help
diff --git a/arch/arm/mach-imx/Makefile b/arch/arm/mach-imx/Makefile
index 27f1976..95925f2 100644
--- a/arch/arm/mach-imx/Makefile
+++ b/arch/arm/mach-imx/Makefile
@@ -104,7 +104,7 @@ obj-$(CONFIG_SMP) += headsmp.o platsmp.o
 obj-$(CONFIG_HOTPLUG_CPU) += hotplug.o
 obj-$(CONFIG_SOC_IMX6Q) += clk-imx6q.o mach-imx6q.o
 obj-$(CONFIG_SOC_IMX6SL) += clk-imx6sl.o mach-imx6sl.o
-obj-$(CONFIG_SOC_IMX6SX) += clk-imx6sx.o mach-imx6sx.o mu.o
+obj-$(CONFIG_SOC_IMX6SX) += clk-imx6sx.o mach-imx6sx.o
 
 ifeq ($(CONFIG_SUSPEND),y)
 AFLAGS_suspend-imx6.o :=-Wa,-march=armv7-a
diff --git a/arch/arm/mach-imx/pm-imx6.c b/arch/arm/mach-imx/pm-imx6.c
index 03f0ed2..2a32e79 100644
--- a/arch/arm/mach-imx/pm-imx6.c
+++ b/arch/arm/mach-imx/pm-imx6.c
@@ -661,9 +661,8 @@ static int imx6q_pm_enter(suspend_state_t state)
 	static unsigned int ccm_ccgr4, ccm_ccgr6;
 
 	if (imx_src_is_m4_enabled()) {
-		if (imx_gpc_is_m4_sleeping() && imx_mu_is_m4_in_low_freq()) {
+		if (imx_gpc_is_m4_sleeping()) {
 			imx_gpc_hold_m4_in_sleep();
-			imx_mu_enable_m4_irqs_in_gic(true);
 		} else {
 			pr_info("M4 is busy, enter WAIT mode instead of STOP!\n");
 			imx6q_set_lpm(WAIT_UNCLOCKED);
@@ -764,7 +763,6 @@ static int imx6q_pm_enter(suspend_state_t state)
 	}
 
 	if (imx_src_is_m4_enabled()) {
-		imx_mu_enable_m4_irqs_in_gic(false);
 		imx_gpc_release_m4_in_sleep();
 	}
 
-- 
2.7.4

